src/librustc/middle/const_limit.rs

```diff
@@ -0,0 +1,26 @@
+
+use crate::session::Session;
+use syntax::ast;
+use syntax::symbol::{Symbol, sym};
+
+use rustc_data_structures::sync::Once;
+
+pub fn update_const_limit(sess: &Session, krate: &ast::Crate) {
+    update_limit(krate, &sess.const_limit, sym::const_limit, 128);
+}
+
+fn update_limit(krate: &ast::Crate, limit: &Once<usize>, name: Symbol, default: usize) {
+    for attr in &krate.attrs {
```

Comment:
I think this would be cleaner as:
```rust
let value = krate
    .attrs
    .iter()
    .filter(|attr| attr.check_name(name))
    .find(|attr| attr.value_str().as_str().parse().ok())
    .unwrap_or(default);

limit.set(value);

Reply:
discussion regarding the possibility to merge `const_limit.rs` and the existing `recursion_limit.rs` is below.

Reply:
Sure; If the identical logic exists in `recursion_limit.rs` I would rewrite that one as well, and try to unify the parsing.